core/src/test/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorServiceSpec.scala

```diff
@@ -738,16 +738,35 @@ class CollectorServiceSpec extends Specification {
     }
 
     "accessControlAllowOriginHeader" in {
+      val origin1 = HttpOrigin("http", Host("origin1"))
       "give a restricted ACAO header if there is an Origin header in the request" in {
-        val origin  = HttpOrigin("http", Host("origin"))
-        val request = HttpRequest().withHeaders(`Origin`(origin))
-        service.accessControlAllowOriginHeader(request) shouldEqual
-          `Access-Control-Allow-Origin`(HttpOriginRange.Default(List(origin)))
+        val request = HttpRequest().withHeaders(`Origin`(origin1))
+        val header = `Access-Control-Allow-Origin`(HttpOriginRange.Default(List(origin1)))
+        "and no domain restrictions" in {
+          service.accessControlAllowOriginHeader(request, None) shouldEqual List(header)
+        }
+        "and a domain restriction which permits that origin" in {
+          service.accessControlAllowOriginHeader(request, Some(List("origin2", "origin1"))) shouldEqual List(header)
+        }
       }
       "give an open ACAO header if there are no Origin headers in the request" in {
         val request = HttpRequest()
-        service.accessControlAllowOriginHeader(request) shouldEqual
-          `Access-Control-Allow-Origin`(HttpOriginRange.`*`)
+        val header = `Access-Control-Allow-Origin`(HttpOriginRange.`*`)
+        "and no domain restrictions" in {
+          service.accessControlAllowOriginHeader(request, None) shouldEqual List(header)
+        }
+        "and any domain restrictions" in {
```

Comment:
Is this what we want? If there are domain restrictions but no `Origin` header, should we not still enforce the restrictions?

Reply:
I slightly agree.  Having said that, this is testing a non-CORS request so really it doesn't matter what ACAO header is in the response.